### ISSUE TITLE
Add unit tests for reservoir environment

### ISSUE THREAD

**Issue by CooperBigFoot** [CooperBigFoot/RL-demo#4]
## Description
Create comprehensive unit tests for both the reservoir simulator and TorchRL environment wrapper to ensure correctness.

## Reference
See planning document: `/scratchpads/phase1_environment_plan.md` - Section 5

## Dependencies
- #2 (Reservoir simulator)
- #3 (TorchRL environment wrapper)

## Tasks
- [ ] Create `tests/test_reservoir_simulator.py`:
  - Test water balance equation
  - Test safety constraints
  - Test inflow generation
  - Test episode length (365 days)
  - Test termination conditions
- [ ] Create `tests/test_discrete_reservoir_env.py`:
  - Test reset/step cycle
  - Test action space validation (all 11 actions)
  - Test state shape and normalization
  - Test reward calculation
  - Test TorchRL spec compliance
  - Test with random policy
- [ ] Add integration test:
  - Test environment with TorchRL's SyncDataCollector
  - Verify TensorDict format
  - Check batch dimension handling

## Test Examples
```python
def test_action_constraints():
    """Verify all discrete actions map to valid releases"""
    env = DiscreteReservoirEnv(config)
    td = env.reset()
    
    for action in range(11):
        td["action"] = torch.tensor(action)
        td = env.step(td)
        # Verify release is within constraints
        assert 0 <= env.last_release <= 0.10 * env.simulator.current_volume

def test_torchrl_specs():
    """Test environment specs are correct"""
    env = DiscreteReservoirEnv(config)
    assert env.observation_spec.shape == (13,)
    assert env.action_spec.n == 11
```

## Acceptance Criteria
- All tests pass
- Code coverage > 90% for environment code
- Tests run in < 10 seconds
- Environment is validated for TorchRL compatibility

## Priority
High - Must validate environment before training

**Comment by CooperBigFoot** [comment]
Tests are already comprehensive enough. Closing as complete.
